fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -58,6 +59,51 @@ where
         })
     }
 
+    /// Performs initialization on the syncer. Requires a minimum of `cached_blocks` to be filled
+    /// in cache first before syncing can be performed.
+    pub async fn init(&mut self, cached_blocks: BlockHeight) -> anyhow::Result<()> {
+        let start = Instant::now();
+
+        let (mut latest_height, mut non_null_parent_hash) = self.latest_cached_data().await;
+        let target_height = cached_blocks + latest_height + 1;
+        tracing::debug!(
+            target_height,
+            latest_height,
+            "syncer init with last committed finality"
+        );
+
+        loop {
+            // obtain the finalized chain head first
```

Comment:
It would be helpful to leave a comment to whether this is going to the parent API or the local ledger.